lib/matplotlib/axes/_base.py

```diff
@@ -2094,22 +2095,36 @@ def autoscale(self, enable=True, axis='both', tight=None):
             The *tight* setting is retained for future autoscaling
             until it is explicitly changed.
 
+        This returns a context-manager/decorator that allows temporarily
+        setting the autoscale status, i.e.::
 
-        Returns None.
+            with axes.autoscale(enable): ...
+
+        or::
+
+            @axes.autoscale(enable)
```

Comment:
I would not suggest this in the documentation as this binds the definition of this function to a specific instance.